core/src/com/unciv/logic/GameInfo.kt

```diff
@@ -260,16 +262,70 @@ class GameInfo {
                 changeBuildingName(cityConstructions, "Machu Pichu", "Machu Picchu")
                 // As of 2.16.1, changed Colloseum to Colosseum
                 changeBuildingName(cityConstructions, "Colloseum", "Colosseum")
+
+                // As of 3.4.9 cities have a unique identifier
+                if (cityInfo.id == "") cityInfo.id = UUID.randomUUID().toString()
             }
 
             // This doesn't HAVE to go here, but why not.
             // As of version 3.1.3, trade offers of "Declare war on X" and "Introduction to X" were changed to X,
             //   with the extra text being added only on UI display (solved a couple of problems).
-            for(trade in civInfo.tradeRequests.map { it.trade })
-                for(offer in trade.theirOffers.union(trade.ourOffers)){
+            for(trade in civInfo.tradeRequests.map { it.trade }) {
+                for (offer in trade.theirOffers.union(trade.ourOffers)) {
                     offer.name = offer.name.removePrefix("Declare war on ")
                     offer.name = offer.name.removePrefix("Introduction to ")
                 }
+            }
+
+            // As of 3.4.9 cities are referenced by id, not by name
+            // So try to update every tradeRequest (if there are no conflicting names)
+            for(tradeRequest in civInfo.tradeRequests) {
+                val trade = tradeRequest.trade
+                val toRemove = ArrayList<TradeOffer>()
+                for (offer in trade.ourOffers) {
+                    if (offer.type == TradeType.City) {
+                        val countNames = civInfo.cities.count { it.name == offer.name}
+
+                        if (countNames == 1)
+                            offer.name = civInfo.cities.first { it.name == offer.name}.id
```

Comment:
I propose to change `offer.name` to `offer.cityId` to remove missunderstanding in future

Reply:
Oh, I thought it used for city name only
Yes, 'value' is good point

Reply:
I reverted the change from `name` to `value` to avoid a migration step only for this renaming.